### ISSUE TITLE
Cannot pass headers as command arguments

### ISSUE THREAD

**Issue by jonhoo** [adnanh/webhook#106]
If I set `pass-arguments-to-command` to include

``` json
{
  "source": "entire-headers"
}
```

or

``` json
{
  "source": "header",
  "name": "X-GitHub-Event"
}
```

The resulting argument is always empty. Can headers only be referenced in `match`, or is the intention that they should also work in arguments?

**Comment by adnanh** [comment]
Hey there, what version of webhook are you using and what is your hooks.json definition for the given hook?

Here is a sample hooks.json definition:
```json
[
  {
    "pass-arguments-to-command": [
      {
        "source": "entire-headers"
      }
    ],
    "execute-command": "/home/adnan/test.sh",
    "id": "test"
  }
]
```

test.sh script:
```sh
#!/bin/sh

echo $1 > /home/adnan/test.log
```

Trigger with curl:
```sh
curl localhost:9000/hooks/test
```

Webhook log after triggering the hook:
```
[webhook] 2016/10/31 13:31:51 version 2.5.0 starting
[webhook] 2016/10/31 13:31:51 setting up os signal watcher
[webhook] 2016/10/31 13:31:51 attempting to load hooks from hooks.json
[webhook] 2016/10/31 13:31:51 found 1 hook(s) in file
[webhook] 2016/10/31 13:31:51 	loaded: test
[webhook] 2016/10/31 13:31:51 serving hooks on http://0.0.0.0:9000/hooks/{id}
[webhook] 2016/10/31 13:31:51 os signal watcher ready
[webhook] 2016/10/31 13:32:24 Started GET /hooks/test
[webhook] 2016/10/31 13:32:24 test got matched
[webhook] 2016/10/31 13:32:24 test hook triggered successfully
[webhook] 2016/10/31 13:32:24 Completed 200 OK in 44.258µs
[webhook] 2016/10/31 13:32:24 executing /home/adnan/test.sh (/home/adnan/test.sh) with arguments ["/home/adnan/test.sh" "{\"Accept\":\"*/*\",\"User-Agent\":\"curl/7.35.0\"}"] and environment [] using  as cwd
[webhook] 2016/10/31 13:32:24 command output: 
[webhook] 2016/10/31 13:32:24 finished handling test
```

Contents of test.log:
```json
{"Accept":"*/*","User-Agent":"curl/7.35.0"}
```

**Comment by adnanh** [comment]
Looking at your script in the commit that referenced this issue, $7 will be JSON string that contains all headers that have been passed to the webhook, it will never be equal to "ping". If you wanted the value from the header that contains ping ("X-Github-Event"?), you should reference it using the syntax you wrote:
```json
{
  "source": "header",
  "name": "X-GitHub-Event"
}
```

**Comment by jonhoo** [comment]
I tried
```json
{
  "source": "header",
  "name": "X-GitHub-Event"
}
```
first, and got an empty string. I then tried
```json
{
  "source": "entire-headers"
}
```
just to see if that would work. It also gives me an empty string.

**Comment by adnanh** [comment]
Weird. Could you create the setup I used in the comment above and copy the webhook output?

**Comment by adnanh** [comment]
Also, what version of webhook are you using?

**Comment by adnanh** [comment]
And which OS is webhook running on?

**Comment by jonhoo** [comment]
```
[webhook] 2016/10/31 13:13:56 Started POST /hooks/email-pushbot
[webhook] 2016/10/31 13:13:56 email-pushbot got matched
[webhook] 2016/10/31 13:13:56 email-pushbot hook triggered successfully
[webhook] 2016/10/31 13:13:56 error extracting command arguments: couldn't retrieve argument for {Source:header Name:X-GitHub-Event EnvName:}
[webhook] 2016/10/31 13:13:56 executing /usr/bin/mailer (/usr/bin/mailer) with arguments ["/usr/bin/mailer" "/srv/git/pushbot" "refs/heads/master" "6db40f5186486801785a24241ffaf1a6ba0d2e1a" "328d298f0a8a44e8653430857eebf5d139c756b7" "jonhoo" "jon@thesquareplanet.com" ""] and environment [] using /home/default as cwd
[webhook] 2016/10/31 13:13:56 command output: handling  to /srv/git/pushbot
/srv/git/pushbot refs/heads/master 6db40f5186486801785a24241ffaf1a6ba0d2e1a 328d298f0a8a44e8653430857eebf5d139c756b7 jonhoo jon@thesquareplanet.com
pushed by: jonhoo <jon@thesquareplanet.com>
aliased to Jon Gjengset <jon@thesquareplanet.com>
syncing with upstream
  Fetching origin
invoking post-receive-email with
  6db40f5186486801785a24241ffaf1a6ba0d2e1a
  328d298f0a8a44e8653430857eebf5d139c756b7
  refs/heads/master
all done
[webhook] 2016/10/31 13:13:56 finished handling email-pushbot
[webhook] 2016/10/31 13:13:56 Completed 200 OK in 392.731546ms
```
```console
$ git -C go/src/github.com/adnanh/webhook/ rev-parse HEAD
7d525cf317b7f0e45b9e7e1a1658b4633154ec01
```
Running Arch Linux.

**Comment by adnanh** [comment]
Aha! Finally got it!

From RFC 2616 - "Hypertext Transfer Protocol -- HTTP/1.1", Section 4.2, "Message Headers":

> Each header field consists of a name followed by a colon (":") and the field value. *Field names are case-insensitive.*

What Golang http package does is normalizes the keys in the request headers map using the "Title case", making webhook unable to extract X-Git*Hub*-Event because in the hash map it's under X-Git*hub*-Event.
As a temporary workaround you can use the X-Github-Event as a header name.

Good catch, thanks!

2.5.1 coming right up!

**Comment by adnanh** [comment]
Fixed in  [2.6.0](https://github.com/adnanh/webhook/releases/tag/2.6.0).

**Comment by jonhoo** [comment]
Great, thank you!
